odl/tomo/operators/xray_trafo.py

```diff
@@ -0,0 +1,285 @@
+            Discretized space, the domain of the forward projector
+        geometry : `Geometry`
+            The geometry of the transform, contains information about
+            the operator range. It needs to have a sampling grid for
+            motion and detector parameters.
+        backend : {'astra_cuda', 'astra_cpu'}, optional
+            Implementation back-end for the transform. Supported back-ends:
+            'astra_cpu': ASTRA toolbox using CPU, only 2D
+            'astra_cuda': ASTRA toolbox, using CUDA, 2D or 3D
+            Default: 'astra'
+        interp : {'nearest', 'linear'}
+            Interpolation type for the discretization of the operator range.
+            Default: 'nearest'
+        """
+        if not isinstance(discr_domain, DiscreteLp):
+            raise TypeError('discretized domain {!r} is not a `DiscreteLp`'
+                            ' instance.'.format(discr_domain))
+
+        if not isinstance(geometry, Geometry):
+            raise TypeError('geometry {!r} is not a `Geometry` instance.'
+                            ''.format(geometry))
+
+        backend = str(backend).lower()
+        if backend not in _SUPPORTED_BACKENDS:
+            raise ValueError('backend {!r} not supported.'
+                             ''.format(backend))
+
+        if backend.startswith('astra'):
+            if not ASTRA_AVAILABLE:
+                raise ValueError('ASTRA backend not available.')
+            if not ASTRA_CUDA_AVAILABLE and backend == 'astra_cuda':
+                raise ValueError('ASTRA CUDA backend not available.')
+            if discr_domain.dspace.dtype not in (np.float32, np.complex64):
+                raise ValueError('ASTRA support is limited to `float32` for '
+                                 'real and `complex64` for complex data.')
+            if not np.allclose(discr_domain.grid.stride[1:],
+                               discr_domain.grid.stride[:-1]):
+                raise ValueError('ASTRA does not support different voxel '
+                                 'sizes per axis (got {}).'
+                                 ''.format(discr_domain.grid.stride))
+
+        self._geometry = geometry
+        self._backend = backend
+        self.kwargs = kwargs
+
+        # Create a discretized space (operator range) with the same data-space
+        # type as the domain.
+        # TODO: maybe use a ProductSpace structure
+        range_uspace = FunctionSpace(geometry.params)
+
+        # Approximate cell volume
+        extent = float(geometry.grid.extent().prod())
+        size = float(geometry.grid.size)
+        weight = extent / size
+
+        range_dspace = discr_domain.dspace_type(
+            geometry.grid.size, weight=weight, dtype=discr_domain.dspace.dtype)
+
+        range_interp = kwargs.get('interp', 'nearest')
+
+        discr_range = DiscreteLp(
+            range_uspace, geometry.grid, range_dspace,
+            interp=range_interp, order=geometry.grid.order)
```

Comment:
Any reason the range should have the same order as the domain? Perhaps this should be documented.